chisel/dataflow_building_blocks/src/main/scala/VectorAdd.scala

```diff
@@ -154,13 +154,13 @@ class VecAddMulDP[T <: UInt](gen : T, vecLen : Int)(implicit params : AccParams)
     })
     
     override def func = {
-      val add = Vec.tabulate(vecLen) {i => io.in1.bits(i) + io.in2.bits(i) }
-      val mul1 = Vec.tabulate(vecLen) {i => io.in1.bits(i) * add(i) }
-      val mul2 = Vec.tabulate(vecLen) {i => io.in2.bits(i) * add(i) }
-      val add2 = Vec.tabulate(vecLen) {i => mul1(i) + mul2(i) }
-      val mul3 = Vec.tabulate(vecLen) {i => mul1(i) * add2(i) }
-      val mul4 = Vec.tabulate(vecLen) {i => mul2(i) * add2(i) }
-      val add3 = Vec.tabulate(vecLen) {i => mul3(i) + mul4(i) }
+      val add = VecInit.tabulate(vecLen) {i => io.in1.bits(i) + io.in2.bits(i) }
```

Comment:
Not sure what this syntax really does and whether it is robust. I usually write it as:
VecInit(Seq.tabulate(vecLen) {...})